src/gui/PianoView.cpp

```diff
@@ -684,14 +684,13 @@ void PianoView::focusOutEvent( QFocusEvent * )
  *  After resizing we need to adjust range of scrollbar for not allowing
  *  to scroll too far to the right.
  *
- *  \param _event resize-event object (unused)
+ *  \param event resize-event object (unused)
  */
-void PianoView::resizeEvent( QResizeEvent * _event )
+void PianoView::resizeEvent(QResizeEvent* event)
 {
-	QWidget::resizeEvent( _event );
-	m_pianoScroll->setRange( 0, WhiteKeysPerOctave * NumOctaves -
-					(int) ceil( (float) width() /
-							PW_WHITE_KEY_WIDTH ) );
+	QWidget::resizeEvent(event);
+	m_pianoScroll->setRange(0, WhiteKeysPerOctave * (NumKeys + 1) / KeysPerOctave -
+					(int)ceil((float)width() / PW_WHITE_KEY_WIDTH));
```

Comment:
Does it have any other benefits apart from being "new and cool"? :) I remember hearing something of the sort that static casting could be faster? Or maybe I'm mistaking it for something else. Performance probably does not matter here though, resize event is not called very often.

Reply:
Since we're writing C++, we want to use C++ casts over C casts.

Benefits are that it tells the compiler exactly what you're trying to do, whereas with C casts the compiler tries whatever it can do to change the type (generally). They help catch errors at compile-time, and are easier to search for and maintain.

Other than `dynamic_cast` they are the same performance-wise.

Reply:
So I went to take a look at the documentation http://www.cplusplus.com/doc/tutorial/typecasting/ and I still have no clue which one I should use with plain old data like here. Basically all they talk about are advantages and applications to pointers to classes.

I come from C on embedded systems background, so I never actually used C++ casts before, and in this case I don't really see how they would help anything -- it would only make the line longer and the code harder to read.

Reply:
This link might explain a little better: https://stackoverflow.com/questions/103512/why-use-static-castintx-instead-of-intx.





> Since we're writing C++, we want to use C++ casts over C casts.
 
as @Veratil has stated this is a c++ project, hence c++ should be used in preference over c style code, even if the c style code still compiles and runs, mixing c and c++ reduces the readability

Reply:
OK, thanks for the explanations, I'll try to change it to static_cast then.

C++ added a lot of stuff and sometimes it is not clear when something is intended to completely replace old "C-like" syntax and when it is just an addition to make life easier or to help solving some specific issues. For example, you wouldn't use lambda functions over regular functions just because they are new in C++. With casting I felt the same way, that the new casts are just for some specific use-cases.